File: ruby/v2.1/create_creative_field_value.rb
#!/usr/bin/env ruby
# Encoding: utf-8
#
# Copyright:: Copyright 2015, Google Inc. All Rights Reserved.
#
# License:: Licensed under the Apache License, Version 2.0 (the "License");
#           you may not use this file except in compliance with the License.
#           You may obtain a copy of the License at
#
#           http://www.apache.org/licenses/LICENSE-2.0
#
#           Unless required by applicable law or agreed to in writing, software
#           distributed under the License is distributed on an "AS IS" BASIS,
#           WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
#           implied.
#           See the License for the specific language governing permissions and
#           limitations under the License.
#
# This example creates a creative field value for a given creative field.
#
# To get the creative field ID, run get_creative_fields.rb.

require_relative 'dfareporting_utils'

def create_creative_field_value(dfareporting, args)
  # Create a new creative field value resource to insert
  field_value = {
    :value => 'Example Creative Field Value'
  }

  # Insert the creative field value
  result = dfareporting.creative_field_values.insert(
    :creativeFieldId => args[:field_id],
    :profileId => args[:profile_id]
  ).body(field_value).execute()

  # Display results
  puts 'Created creative field value with ID %d and value "%s".' %
      [result.data.id, result.data.value]
end

if __FILE__ == $0
  # Retrieve command line arguments
  args = DfaReportingUtils.get_arguments(ARGV, :profile_id, :field_id)

  # Authenticate and initialize API service
  dfareporting = DfaReportingUtils.setup()

  create_creative_field_value(dfareporting, args)
end

File: python/v3_5/create_creative_field_value.py
#!/usr/bin/python
#
# Copyright 2015 Google Inc. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

"""This example creates a creative field value for a given creative field.

To get the creative field ID, run get_creative_fields.py.
"""

import argparse
import sys

import dfareporting_utils
from oauth2client import client

# Declare command-line flags.
argparser = argparse.ArgumentParser(add_help=False)
argparser.add_argument(
    'profile_id', type=int,
    help='The ID of the profile to add a creative field value for')
argparser.add_argument(
    'field_id', type=int,
    help='The ID of the creative field to add a value for')


def main(argv):
  # Retrieve command line arguments.
  flags = dfareporting_utils.get_arguments(argv, __doc__, parents=[argparser])

  # Authenticate and construct service.
  service = dfareporting_utils.setup(flags)

  profile_id = flags.profile_id
  field_id = flags.field_id

  try:
    # Construct and save creative field value
    field_value = {
        'value': 'Test Creative Field Value'
    }

    request = service.creativeFieldValues().insert(
        profileId=profile_id, creativeFieldId=field_id, body=field_value)

    # Execute request and print response.
    response = request.execute()

    print ('Created creative field value with ID %s and value "%s".'
           % (response['id'], response['value']))

  except client.AccessTokenRefreshError:
    print ('The credentials have been revoked or expired, please re-run the '
           'application to re-authorize')


if __name__ == '__main__':
  main(sys.argv)
